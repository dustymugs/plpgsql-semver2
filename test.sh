#!/bin/bash

if [ -z "$POSTGRES_SEARCH_PATH" ]; then
	POSTGRES_SEARCH_PATH="public"
fi

for sql in $(find ./tests -name test.*.sql); do
	rm -rf /tmp/semver.out
	expected=${sql/%.sql/.expected}
	echo -n $sql "..."
	psql "options=--search_path=${POSTGRES_SEARCH_PATH}" -q -A -t -f "$sql"	> /tmp/semver.out
	diffs=$(diff /tmp/semver.out "$expected")
	if [ -z "$diffs" ]; then
		echo "OK"
	else
		echo $diffs
	fi
done

rm -rf /tmp/semver.out
