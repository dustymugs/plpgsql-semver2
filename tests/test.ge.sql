SELECT (to_semver('1.2.3') >= to_semver('1.2.2')); -- TRUE
SELECT (to_semver('1.2.3') >= to_semver('1.2.3')); -- TRUE
SELECT (to_semver('1.2.3') >= to_semver('1.2.4')); -- FALSE
SELECT (to_semver('1.2.3') >= to_semver('1.2.3-beta')); -- TRUE
SELECT to_semver('1.0.0-alpha.1') >= to_semver('1.0.0-alpha'); -- TRUE
SELECT to_semver('1.0.0-alpha.beta') >= to_semver('1.0.0-alpha.1'); -- TRUE
SELECT to_semver('1.0.0-beta') >= to_semver('1.0.0-alpha.beta'); -- TRUE
SELECT to_semver('1.0.0-beta.2') >= to_semver('1.0.0-beta'); -- TRUE
SELECT to_semver('1.0.0-beta.11') >= to_semver('1.0.0-beta.2'); -- TRUE
SELECT to_semver('1.0.0-rc.1') >= to_semver('1.0.0-beta.11'); -- TRUE
SELECT to_semver('1.0.0') >= to_semver('1.0.0-rc.1'); -- TRUE
SELECT to_semver('1.0.1') >= to_semver('1.0.0'); -- TRUE
SELECT to_semver('1.0.0-alpha.1.b.2') >= to_semver('1.0.0-alpha.1.b.2'); -- TRUE
SELECT to_semver('1.0.0-alpha.1.b.2') >= to_semver('1.0.0-alpha.1.b.3'); -- FALSE
SELECT to_semver('1.0.0-alpha.1.b') >= to_semver('1.0.0-alpha.1.b.3'); -- FALSE
