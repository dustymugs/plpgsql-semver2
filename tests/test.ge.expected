t
t
f
t
t
t
t
t
t
t
t
t
t
f
f
